### PR TITLE
♻️ update settings api server

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- ♻️ refactors ``api-server`` to use ``settings-library`` (based on pattern presented in http://github.com/pcrespov/sample-app-settings)
- ♻️ formatted ``meta`` constants to uppercase in both ``api-server`` and ``director-v2``
- ✨adds CLI command to resolve and print envfiles
- 🐛added a retry policy to retrieve index from static web server since i noticed that  unhandled exception shut down the web-server in one of the tests:
```
   aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host static-webserver:8000 ssl:default [Name or service not known]
```
## Related issue/s

- Follows from PR #2558
- 

## How to test

- all tests in api-server
- check CLI to produce e.g. *envfiles*  
```
$ cd services/api-server
$ make install-dev
$ simcore-service-api-server --help   
...
$ simcore-service-api-server settings --help
...
$ export $(grep -v '^#' .env-devel | xargs),  simcore-service-api-server settings --compact
```

## Checklist

check and update settings in
- [x] ``core/settings.py``
- [x] ``.env-dev``
- [x] ``docker-compose``

### CODE REVIEW

File: packages/settings-library/src/settings_library/postgres.py [hunk 1]
```diff
@@ -53,6 +53,12 @@ def dsn(self) -> str:
             port=f"{self.POSTGRES_PORT}",
             path=f"/{self.POSTGRES_DB}",
         )
+        return dsn
+
+    @cached_property
+    def dsn_with_query(self) -> str:
+        """Some clients do not support queries in the dsn"""
```
Comment (sanderegg) [inline packages/settings-library/src/settings_library/postgres.py:60]: really? which ones?

**Review by GitHK — APPROVED** [review]

File: services/api-server/.env-devel [hunk 2]
```diff
@@ -20,14 +31,12 @@ SC_BOOT_MODE=debug-ptvsd
 
 
 # webserver
-WEBSERVER_ENABLED=1
 WEBSERVER_HOST=webserver
 # Take from general .env-devel
-WEBSERVER_SESSION_SECRET_KEY=REPLACE ME with a key of at least length 32.
+WEBSERVER_SESSION_SECRET_KEY="REPLACE ME with a key of at least length 32."
```
Comment (sanderegg) [inline services/api-server/.env-devel:36]: 👍 this will help in interviews ;)

File: services/api-server/setup.py [hunk 3]
```diff
@@ -63,4 +64,9 @@ def read_reqs(reqs_path: Path) -> Set[str]:
         test_suite="tests",
         tests_require=TEST_REQUIREMENTS,
         extras_require={"test": TEST_REQUIREMENTS},
+        entry_points={
+            "console_scripts": [
+                "simcore-service-api-server = simcore_service_api_server.cli:main",
```
Comment (sanderegg) [inline services/api-server/setup.py:69]: ok that is what you want me to add in the catalog right?

**Review by sanderegg — APPROVED** [review]
great. thanks for the changes!

File: services/api-server/setup.py (hunk 3 above)
Comment (pcrespov) [reply services/api-server/setup.py:69]: that was my suggestion, yes :-)

File: packages/settings-library/src/settings_library/postgres.py (hunk 1 above)
Comment (pcrespov) [reply packages/settings-library/src/settings_library/postgres.py:60]: See e.g. this  [code](https://github.com/ITISFoundation/osparc-simcore/blob/f91f7ea63f20a9549db6352b25e90ce2099028fc/packages/service-library/src/servicelib/common_aiopg_utils.py#L48) with ``aiopg.sa.create_engine``. the application name cannot be passed as a part of the dsn

**Pull request merged**